openupgradelib/openupgrade.py

```diff
@@ -380,6 +381,89 @@ def rename_columns(cr, column_spec):
             cr.execute('DROP INDEX IF EXISTS "%s_%s_index"' % (table, old))
 
 
+def rename_fields(env, field_spec, no_deep=False):
+    """Rename fields. Typically called in the pre script. WARNING: If using
+    this on base module, pass the argument ``no_deep`` with True value for
+    avoiding the using of the environment (which is not yet loaded).
+
+    This, in contrast of ``rename_columns``, performs all the steps for
+    completely rename a field from one name to another. This is needed for
+    making a complete renaming of a field with all their side features:
+    translations, filters, exports...
+
+    Call this method whenever you are not performing a pure SQL column renaming
+    for other purposes (preserve a value for example).
+
+    This method performs also the SQL column renaming, so only one call is
+    needed.
+
+    :param env: Environment/pool variable. The database cursor is the only
+      thing needed, but added in prevision of TODO tasks for not breaking
+      API later.
+    :param fields_spec: a list of tuples with the following elements:
+      * Model name. The name of the Odoo model
+      * Table name. The name of the SQL table for the model.
+      * Old field name. The name of the old field.
+      * New field name. The name of the new field.
+    :param no_deep: If True, avoids to perform any operation that involves
+      the environment. Not used for now.
+    """
+    cr = env.cr
+    for model, table, old_field, new_field in field_spec:
+        if column_exists(cr, table, old_field):
+            rename_columns(cr, {table: [(old_field, new_field)]})
+        # Rename corresponding field entry
+        cr.execute("""
+            UPDATE ir_model_fields
+            SET name = %s
+            WHERE name = %s
+                AND model = %s
+            """, (new_field, old_field, model),
+        )
+        # Rename translations
+        cr.execute("""
+            UPDATE ir_translation
+            SET name = %s
+            WHERE name = %s
+                AND type = 'model'
+            """, (
+                "%s,%s" % (model, old_field),
+                "%s,%s" % (model, new_field),
+            ),
+        )
+        # Rename appearances on export profiles
+        # TODO: Rename when the field is part of a submodel (ex. m2one.field)
+        cr.execute("""
+            UPDATE ir_exports_line
+            SET name = %s
+            WHERE name = %s
+            """, (old_field, new_field),
+        )
+        # Rename appearances on filters
+        # Example of replaced domain: [['field', '=', self], ...]
+        # TODO: Rename when the field is part of a submodel (ex. m2one.field)
+        cr.execute("""
+            UPDATE ir_filters
+            SET domain = replace(domain, $$'%s'$$, $$'%s'$$)
```

Comment:
You could use `\m%s\M` to match whole words, because in case there's a `id` field, we don't want `partner_id` to match!

Check it out: https://www.postgresql.org/docs/current/static/functions-matching.html#posix-constraint-escapes-table

Reply:
It's not the case for this, @Yajo . See that `'` are surrounding the word.

Reply:
It's expected that using my fix works for full words, no matter if they are surrounded by `'`, `"` or found in a relational field such as `parent_id.matching_field`. Isn't that what you'd want here? 🤔

Reply:
NO, in this case it's a simple substitution. See that we are matching simple domains for the same model, so that's exactly what we want to match. 

The rest is not possible for multiple reasons:

* The model of the filter will be another.
* We have to determine if the submodel is for the model with the replaced field, and not another with a field with the same old name.

That's why the TODO is there, because it can't be resolved simply with a regex.